src/libraries/System.Private.CoreLib/src/System/TypeLoadException.cs

```diff
@@ -29,15 +29,6 @@ public TypeLoadException(string? message, Exception? inner)
             HResult = HResults.COR_E_TYPELOAD;
         }
 
-#if !MONO
-        internal TypeLoadException(string message, string typeName)
```

Comment:
This constructor is used internally by CoreCLR to construct TypeLoadExceptions. If you delete it, you will have to redo how type load exceptions are created internally by CoreCLR.

It would be much less invasive to keep it, enable it for Mono and fix this exclusively on the Mono side.

Reply:
Disabling the failing test on Mono would be fine too.

Reply:
> This constructor is used internally by CoreCLR to construct TypeLoadExceptions. If you delete it, you will have to redo how type load exceptions are created internally by CoreCLR.

I think I redid it? CI is mostly green apart from unrelated failures, and this overload was added three weeks ago in #111374 and there weren't any uses from C++ there.

Reply:
I was worried about the places where the VM creates the exception using "reflection" like: https://github.com/dotnet/runtime/blob/16782a4481f389e68314b16ea628c5af80f22783/src/coreclr/vm/clrex.cpp#L1471-L1472 . A good point that this constructor was added just recently, so it is not used like that. 

Still, I am not sure whether the init-only is an improvement. If we were to introduce a public API to allow creating TypeLoadException with type name, I think we would add a constructor that takes the additional argument for consistency with how all existing BCL exception constructors look.

Reply:
Yes, a `(string message, string typeName)` constructor would be the ideal API shape but that would require changes in Mono's C code, which I would prefer to not expand this PR's scope with.

Reply:
I thought of a (likely hacky) way that both simulates a `(message, typeName)` overload, and avoids changes to the Mono runtime. Let me know what you think.

Reply:
I am ok with it. I left a few comments.